tests/v2/validation/networking/connectivity/port_test.go

```diff
@@ -323,28 +612,160 @@ func (p *PortTestSuite) TestLoadBalance() {
-func getNodePort() int {
-	return rand.IntN(2767) + 30000
+func (p *PortTestSuite) validateHostPortSSH(steveClient *steveV1.Client, hostPort int, workloadName string, namespaceName string) {
+	p.T().Logf("Getting the node using the label [%v]", labelWorker)
+	query, err := url.ParseQuery(labelWorker)
+	assert.NoError(p.T(), err)
+
+	nodeList, err := steveClient.SteveType("node").List(query)
+	assert.NoError(p.T(), err)
+	assert.NotEmpty(p.T(), nodeList.Data)
+
+	_, stevecluster, err := clusters.GetProvisioningClusterByName(p.client, p.cluster.Name, provisioninginput.Namespace)
+
+	wc, err := p.client.WranglerContext.DownStreamClusterWranglerContext(p.cluster.ID)
+	require.NoError(p.T(), err)
+
+	pods, err := wc.Core.Pod().List(namespaceName, metav1.ListOptions{})
+	assert.NoError(p.T(), err)
+	assert.NotEmpty(p.T(), pods)
+
+	var nodes []string
+	nodes = make([]string, 0)
+	for i := 0; i < len(pods.Items); i++ {
+		nodeName := pods.Items[i].Spec.NodeName
+		nodes = append(nodes, nodeName)
+	}
+	assert.NotEmpty(p.T(), nodes)
+
+	for _, machine := range nodeList.Data {
+		p.T().Log("Getting the node IP")
+		newNode := &corev1.Node{}
+		err = steveV1.ConvertToK8sType(machine.JSONResp, newNode)
+		assert.NoError(p.T(), err)
+		assert.NotEmpty(p.T(), newNode.Name)
+
+		_, found := slices.BinarySearch(nodes, newNode.Name)
+		if found {
+			// Project Network Isolation should be enabled when setting up the cluster for this test
+			nodeIP := kubeapinodes.GetNodeIP(newNode, corev1.NodeInternalIP)
+
+			sshUser, err := sshkeys.GetSSHUser(p.client, stevecluster)
+			assert.NoError(p.T(), err)
+			assert.NotEmpty(p.T(), sshUser, errors.New("sshUser does not exist"))
+
+			sshNode, err := sshkeys.GetSSHNodeFromMachine(p.client, sshUser, &machine)
+			assert.NoError(p.T(), err)
+
+			log, err := sshNode.ExecuteCommand(fmt.Sprintf("curl %s:%s/name.html", nodeIP, strconv.Itoa(hostPort)))
+			if err != nil && !errors.Is(err, &ssh.ExitMissingError{}) {
+				assert.NoError(p.T(), err)
+			}
+			p.T().Logf("Log of the curl command {%v}", log)
+			require.True(p.T(), strings.Contains(log, workloadName))
+		}
+	}
+}
+
+func (p *PortTestSuite) validateLoadBalance(steveClient *steveV1.Client, nodePort int, workloadName string) {
+	p.T().Logf("Getting the node using the label [%v]", labelWorker)
+	query, err := url.ParseQuery(labelWorker)
+	assert.NoError(p.T(), err)
+
+	nodeList, err := steveClient.SteveType("node").List(query)
+	assert.NoError(p.T(), err)
+	assert.NotEmpty(p.T(), nodeList.Data)
+
+	for _, machine := range nodeList.Data {
+		p.T().Log("Getting the node IP")
+		newNode := &corev1.Node{}
+		err = steveV1.ConvertToK8sType(machine.JSONResp, newNode)
+		assert.NoError(p.T(), err)
+
+		nodeIP := kubeapinodes.GetNodeIP(newNode, corev1.NodeExternalIP)
+		if nodeIP == "" {
+			nodeIP = kubeapinodes.GetNodeIP(newNode, corev1.NodeInternalIP)
+		}
+
+		log, err := curlCommand(p.client, p.cluster.ID, fmt.Sprintf("%s:%s/name.html", nodeIP, strconv.Itoa(nodePort)))
```

Comment:
nit; the log from this func doesn't seem very helpful. Maybe you could cat the file or something as the log? As-is, it just shows the download status of the curl, which jumbles the logs a bit. (at least I think its from here. Log starting with 
```
Log of the curl command curl ...
```